engine/enqueue.go

```diff
@@ -26,16 +26,18 @@ type Pattern struct {
 	Match MatchFunc
 	// Map is a function to apply to messages before storing them. If not provided, then the message is stored in its original form.
 	Map MapFunc
+	// Validate is a function to validate a message before storing it. If not provided, then the message is always stored.
+	Validate ValidateFunc
```

Comment:
I would suggest to document the kind of validation we expect to happen here, and the expected error returns. In particular:
* should only include cheap validation checks (else we're blocking networking layer threads again)
* if a message passes validation here, it may still be invalid (validation is not exhaustive)
* either define an expected sentinel error type, or "all generic errors are benign and indicate an invalid message"